crates/hir_ty/src/tests/regression.rs

```diff
@@ -1348,21 +1348,161 @@ fn test() {
         "#,
         expect![[r#"
         67..68 'x': LD
-        94..138 '{     ...   r }': LinkArray<LD, {unknown}>
-        104..105 'r': LinkArray<LD, {unknown}>
-        108..126 'LinkAr...LD, N>': LinkArray<LD, {unknown}>(LD) -> LinkArray<LD, {unknown}>
-        108..129 'LinkAr... N>(x)': LinkArray<LD, {unknown}>
+        94..138 '{     ...   r }': LinkArray<LD, N>
+        104..105 'r': LinkArray<LD, N>
+        108..126 'LinkAr...LD, N>': LinkArray<LD, N>(LD) -> LinkArray<LD, N>
+        108..129 'LinkAr... N>(x)': LinkArray<LD, N>
         127..128 'x': LD
-        135..136 'r': LinkArray<LD, {unknown}>
+        135..136 'r': LinkArray<LD, N>
         150..232 '{     ...(x); }': ()
-        160..161 'x': LinkArray<i32, {unknown}>
-        164..175 'f::<i32, 2>': fn f<i32, i32>(i32) -> LinkArray<i32, {unknown}>
-        164..178 'f::<i32, 2>(5)': LinkArray<i32, {unknown}>
+        160..161 'x': LinkArray<i32, 2>
+        164..175 'f::<i32, 2>': fn f<2, i32>(i32) -> LinkArray<i32, 2>
+        164..178 'f::<i32, 2>(5)': LinkArray<i32, 2>
         176..177 '5': i32
-        188..189 'y': LinkArray<LinkArray<i32, {unknown}>, {unknown}>
-        192..226 'LinkAr...>, 52>': LinkArray<LinkArray<i32, {unknown}>, {unknown}>(LinkArray<i32, {unknown}>) -> LinkArray<LinkArray<i32, {unknown}>, {unknown}>
-        192..229 'LinkAr...52>(x)': LinkArray<LinkArray<i32, {unknown}>, {unknown}>
-        227..228 'x': LinkArray<i32, {unknown}>
+        188..189 'y': LinkArray<LinkArray<i32, 2>, 52>
+        192..226 'LinkAr...>, 52>': LinkArray<LinkArray<i32, 2>, 52>(LinkArray<i32, 2>) -> LinkArray<LinkArray<i32, 2>, 52>
+        192..229 'LinkAr...52>(x)': LinkArray<LinkArray<i32, 2>, 52>
+        227..228 'x': LinkArray<i32, 2>
         "#]],
     );
 }
+
+#[test]
+fn const_generic_error_telorance() {
+    check_infer(
```

Comment:
as a general note, I kind of want to get rid of `check_infer` tests. More targeted assertions are more useful in most cases.

Reply:
I just want them to not panic, so I added a `check_all` that calls `infer` and ignore its output (`check_types` seems doesn't catch every panic) Is it ok?

Reply:
I think `check_no_mismatches` should work fine for that. Not sure what panics that wouldn't trigger that `infer` does.